### PR TITLE
rpc: return protocol parameters in `getversion`, fix #2160

### LINKED ISSUE #2160: Extend getversion RPC with protocol data
See neo-project/neo-modules#639.

### PR DESCRIPTION
**Pull request by fyrchik**
`StateRootInHeader` is duplicated similarly to `Network`.
It will be removed in future as it is surely a protocol parameter.

Signed-off-by: Evgeniy Stratonikov <evgeniy@nspcc.ru>

### CODE REVIEW

File: pkg/rpc/response/result/version.go [hunk 1]
```diff
@@ -11,6 +14,22 @@ type (
 		WSPort    uint16        `json:"wsport,omitempty"`
 		Nonce     uint32        `json:"nonce"`
 		UserAgent string        `json:"useragent"`
+		Protocol  Protocol      `json:"protocol"`
+		// StateRootInHeader is true if state root is contained in block header.
+		// Deprecated: use Protocol.StateRootInHeader instead
```
Comment (roman-khimov) [inline pkg/rpc/response/result/version.go:19]: `Magic` is deprecated too.

File: pkg/rpc/client/client.go [hunk 2]
```diff
@@ -119,7 +119,7 @@ func (c *Client) Init() error {
 		return fmt.Errorf("failed to get network magic: %w", err)
 	}
 	c.network = version.Magic
-	c.stateRootInHeader = version.StateRootInHeader
+	c.stateRootInHeader = version.Protocol.StateRootInHeader
```
Comment (roman-khimov) [inline pkg/rpc/client/client.go:122]: Both of the values should check for `Protocol.X` first and then use values from old locations if there is nothing there.

File: pkg/rpc/response/result/version.go [hunk 3]
```diff
@@ -1,16 +1,37 @@
...
+		StateRootInHeader bool `json:"staterootinheader,omitempty"`
+	}
+
+	// Protocol represents network-dependent parameters.
+	Protocol struct {
+		AddressVersion              byte          `json:"addressversion"`
+		Network                     netmode.Magic `json:"network"`
+		MillisecondsPerBlock        int           `json:"msperblock"`
+		MaxTraceableBlocks          uint32        `json:"maxtraceableblock"`
```
Comment (AnnaShaleva) [inline pkg/rpc/response/result/version.go:30]: `json:"maxtraceableblocks"`

**Review by AnnaShaleva — CHANGES_REQUESTED** [review]

File: pkg/rpc/response/result/version.go (hunk 3 above)
Comment (roman-khimov) [reply pkg/rpc/response/result/version.go:30]: @fyrchik?

File: pkg/rpc/client/client.go [hunk 4]
```diff
@@ -118,8 +118,14 @@ func (c *Client) Init() error {
 	if err != nil {
 		return fmt.Errorf("failed to get network magic: %w", err)
 	}
-	c.network = version.Magic
-	c.stateRootInHeader = version.StateRootInHeader
+	c.network = version.Protocol.Network
+	if c.network == 0 {
```
Comment (roman-khimov) [inline pkg/rpc/client/client.go:122]: It technically can be 0. But `MillisecondsPerBlock` can't (if we can't detect `Protocol` presence in any other way).

File: pkg/rpc/response/result/version.go (hunk 3 above)
Comment (fyrchik) [reply pkg/rpc/response/result/version.go:30]: Fixed.

File: pkg/rpc/client/client.go (hunk 4 above)
Comment (fyrchik) [reply pkg/rpc/client/client.go]: Fixed.

File: pkg/rpc/response/result/version.go (hunk 3 above)
Comment (AnnaShaleva) [reply pkg/rpc/response/result/version.go:30]: @fyrchik, you might've forgotten to update PR, not fixed yet.

Comment (fyrchik) [reply pkg/rpc/response/result/version.go]: accidentally changed `maxtransactionperblock`, fixed now.

**Review by roman-khimov — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**